libshaderc_spvc/src/spvcir_pass.cc

```diff
@@ -14,25 +14,189 @@
+      switch (lang) {
+        case spv::SourceLanguageESSL:
+          ir_->source.es = true;
+          ir_->source.version = inst->GetSingleWordOperand(1u);
+          ir_->source.known = true;
+          ir_->source.hlsl = false;
+          break;
+
+        case spv::SourceLanguageGLSL:
+          ir_->source.es = false;
+          ir_->source.version = inst->GetSingleWordOperand(1u);
+          ir_->source.known = true;
+          ir_->source.hlsl = false;
+          break;
+
+        case spv::SourceLanguageHLSL:
+          // For purposes of cross-compiling, this is GLSL 450.
+          ir_->source.es = false;
+          ir_->source.version = 450;
+          ir_->source.known = true;
+          ir_->source.hlsl = true;
+          break;
+
+        default:
+          ir_->source.known = false;
+          break;
+      }
+      break;
+    }
+
+    case SpvOpCapability: {
+      auto cap = inst->GetSingleWordOperand(0u);
+      assert(cap != spv::CapabilityKernel &&
+             "Kernel capability not supported.");
+      ir_->declared_capabilities.push_back(static_cast<spv::Capability>(cap));
+      break;
+    }
+
+    case SpvOpExtension: {
+      const std::string extension_name =
+          reinterpret_cast<const char *>(inst->GetInOperand(0u).words.data());
+      ir_->declared_extensions.push_back(std::move(extension_name));
+      break;
+    }
+
+    case SpvOpMemoryModel: {
+      ir_->addressing_model =
+          static_cast<spv::AddressingModel>(inst->GetSingleWordInOperand(0u));
+      ir_->memory_model =
+          static_cast<spv::MemoryModel>(inst->GetSingleWordInOperand(1u));
+      break;
+    }
+
+    case SpvOpEntryPoint: {
+      auto function_id = inst->GetSingleWordInOperand(1u);
+      auto execution_mode =
+          static_cast<spv::ExecutionModel>(inst->GetSingleWordInOperand(0u));
+
+      const char *entry_name =
+          reinterpret_cast<const char *>(inst->GetInOperand(2u).words.data());
+
+      auto itr = ir_->entry_points.insert(std::make_pair(
+          function_id, spirv_cross::SPIREntryPoint(function_id, execution_mode,
+                                                   entry_name)));
+      auto &e = itr.first->second;
+      ir_->set_name(function_id, e.name);
+
+      for (uint32_t i = 3; i < inst->NumInOperands(); ++i) {
+        e.interface_variables.push_back(inst->GetSingleWordInOperand(i));
+      }
+
+      if (!ir_->default_entry_point) ir_->default_entry_point = function_id;
+      break;
+    }
+
+    case SpvOpTypeVoid: {
+      auto id = inst->result_id();
+      auto &type = set<spirv_cross::SPIRType>(id);
+      type.basetype = spirv_cross::SPIRType::Void;
+      break;
+    }
+
+    case SpvOpTypeFunction: {
+      auto id = inst->result_id();
+      auto result = inst->type_id();
+
+      // reading Parameter 0 Type, Parameter 1 Type, ...
+      auto &func = set<spirv_cross::SPIRFunctionPrototype>(id, result);
+      for (uint32_t i = 2; i < inst->NumInOperands(); ++i) {
```

Comment:
start i at 1 to skip over the function return type parameter.

This will only show itself to be a problem once you start having functions that have parameters. This code will skip the first parameter.